og_ui/src/Plugin/Field/FieldFormatter/GroupSubscribeFormatter.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og_ui\Plugin\Field\FieldFormatter\GroupSubscribeFormatter.
+ */
+
+namespace Drupal\og_ui\Plugin\Field\FieldFormatter;
+
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\og\Og;
+use Drupal\user\EntityOwnerInterface;
+
+/**
+ * Plugin implementation for the OG subscribe formatter.
+ *
+ * @FieldFormatter(
+ *   id = "og_ui_group_subscribe",
+ *   label = @Translation("OG Group subscribe"),
+ *   description = @Translation("Display OG Group subscribe links."),
+ *   field_types = {
+ *     "og_group"
+ *   }
+ * )
+ */
+class GroupSubscribeFormatter extends FormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    $entity = $items->getEntity();
+    // @todo Inject this properly.
```

Comment:
Actually, we can't actually do this. We don't get a choice about which services are injected into the formatter plugin instance. It is not container aware :/